camel/agents/chat_agent.py

```diff
@@ -342,6 +329,15 @@ def step(
                 )
                 break
 
+        # Loop over responses terminations
+        if self.response_terminators is not None:
+            for terminator in self.response_terminators:
+                terminated, termination_reason = terminator.is_terminated(
+                    output_messages)
+                if terminated:
+                    for i in range(len(info["termination_reasons"])):
+                        info["termination_reasons"][i] = termination_reason
```

Comment:
How is info["termination_reasons"] initialized?

Reply:
Why do you overwrite the value? This pattern breaks immutability of variables.

Reply:
The code `for i in range(len())` is an anti pattern

Reply:
Moved it before the `self.get_info()` (to avoid breaking the immutability) and make it replace the original finish reason.
Original finish reason is something like `'stop'` which for each message returned from OpenAI api.